src/schemathesis/cli/junitxml.py

```diff
@@ -7,25 +7,23 @@
 
 from ..models import Status
 from ..runner import events
-from .handlers import EventHandler, ExecutionContext
-from .output import default
+from .handlers import EventHandler, ExecutionContext, _get_unique_failures
 
 
-@attr.s(slots=True)
+@attr.s(slots=True)  # pragma: no mutate
 class JunitXMLHandler(EventHandler):
-    file_handle: LazyFile = attr.ib()
-    test_cases: List = attr.ib(factory=list)
-    start_time: Optional[float] = attr.ib(default=None)
+    file_handle: LazyFile = attr.ib()  # pragma: no mutate
+    test_cases: List = attr.ib(factory=list)  # pragma: no mutate
+    start_time: Optional[float] = attr.ib(default=None)  # pragma: no mutate
 
     def handle_event(self, context: ExecutionContext, event: events.ExecutionEvent) -> None:
         if isinstance(event, events.Initialized):
             self.start_time = event.start_time
-        if isinstance(event, events.BeforeExecution):
-            self.test_cases.append(TestCase(f"{event.method} {event.path}"))
         if isinstance(event, events.AfterExecution):
+            self.test_cases.append(TestCase(f"{event.result.method} {event.result.path}"))
```

Comment:
Could you, please, create a variable here and `append` at the end of the branch to avoid that list access at all?